### PR TITLE
track remaining voters and print with tally

### PR DESCRIPTION
**Pull request by birdonfire**
also minor update to event loop logic
- only check if voting is done during day
- only check if night actions are complete during night

### CODE REVIEW

File: lib/werewolf/game.rb [hunk 1]
```diff
@@ -533,7 +543,7 @@ def print_claims
 
     def print_tally
       changed
-      notify_observers(:action => 'tally', :vote_tally => vote_tally)
+      notify_observers(:action => 'tally', :vote_tally => vote_tally, :remaining_votes => remaining_votes)
```
Comment (rue-dhurd) [inline lib/werewolf/game.rb]: Will this make the response too long now?

File: lib/werewolf/game.rb [hunk 2]
```diff
@@ -438,23 +440,31 @@ def assign_roles
...
+    def day?
+      !night?
+    end
 
     def advance_time
       @time_remaining_in_round = default_time_remaining_in_round
       @time_period, @day_number = @time_period_generator.next
 
-      if 'night' == time_period
+      if night?
```
Comment (rue-dhurd) [inline lib/werewolf/game.rb]: I feel like this guy should have a constant.

File: lib/werewolf/game.rb [hunk 3]
```diff
@@ -215,12 +223,7 @@ def night_finished?
 
 
     def detect_voting_finished?
-      (living_players.size == vote_count)
-    end
-
-
-    def vote_count
-      @vote_tally.values.reduce(0) {|count, s| count + s.size}
+      (@remaining_votes.size == 0)
```
Comment (rue-dhurd) [inline lib/werewolf/game.rb]: `<=` for defensive programming? although we'd also be cool with it breaking so we could find and fix.

File: lib/werewolf/game.rb (hunk 2 above)
Comment (ngenator) [inline lib/werewolf/game.rb]: it will! i have a state overhaul pr

File: lib/werewolf/game.rb [hunk 4]
```diff
@@ -438,23 +440,31 @@ def assign_roles
       end
     end
 
+    def night?
+      ('night' == time_period)
+    end
```
Comment (dissemble) [inline lib/werewolf/game.rb]: 👍 thanks for these

File: lib/werewolf/game.rb (hunk 1 above)
Comment (birdonfire) [inline lib/werewolf/game.rb]: too long how?

Comment (rue-dhurd) [inline lib/werewolf/game.rb]: Like the output of `w tally`. I'd want to avoid the 'Show More' text if possible.

File: test/werewolf/game_test.rb [hunk 5]
```diff
@@ -972,31 +1033,6 @@ def test_roles_with_night_actions
     end
 
 
-    def test_vote_count_with_one_candidate
```
Comment (dissemble) [inline test/werewolf/game_test.rb]: why don't we like these?

**Comment by dissemble** [comment]
bump

**Pull request merged**